components/lib/stepper/Stepper.js

```diff
@@ -0,0 +1,334 @@
+import * as React from 'react';
+import { PrimeReactContext } from '../api/Api';
+import { useHandleStyle } from '../componentbase/Componentbase';
+import { CSSTransition } from '../csstransition/CSSTransition';
+import { useMergeProps, useMountEffect } from '../hooks/Hooks';
+import { UniqueComponentId, classNames } from '../utils/Utils';
+import { StepperBase } from './StepperBase';
+import { StepperContent } from './StepperContent';
+import { StepperHeader } from './StepperHeader';
+import { StepperSeparator } from './StepperSeparator';
+
+export const Stepper = React.memo(
+    React.forwardRef((inProps, ref) => {
+        const mergeProps = useMergeProps();
+        const context = React.useContext(PrimeReactContext);
+        const props = StepperBase.getProps(inProps, context);
+        const { ptm, cx, isUnstyled, ptmo } = StepperBase.setMetaData({
+            props
+        });
+        const [idState, setIdState] = React.useState(props.id);
+        const [activeStepState, setActiveStepState] = React.useState(props.activeStep);
+        const navRef = React.useRef();
+
+        useHandleStyle(StepperBase.css.styles, isUnstyled, { name: 'stepper' });
+
+        useMountEffect(() => {
+            if (!idState) {
+                setIdState(UniqueComponentId());
+            }
+        });
+
+        const getStepProp = (step, name) => {
+            return step.props ? step.props[name] : undefined;
+        };
+
+        const getStepKey = (step, index) => {
+            return getStepProp(step, 'header') || index;
+        };
+
+        const isStep = (child) => {
+            return child.type.displayName === 'StepperPanel';
+        };
+
+        const isStepActive = (index) => {
+            return activeStepState === index;
+        };
+
+        const isItemDisabled = (index) => {
+            return props.linear && !isStepActive(index);
+        };
+
+        const updateActiveStep = (event, index) => {
+            setActiveStepState(index);
+
+            props.onChangeStep && props.onChangeStep({ originalEvent: event, index });
```

Comment:
My only thought here is what I have done in other components is give the user a change to say they don't want to change the step because some validation they have or some check.  So someting like this...

```js
const updateActiveStep = (event, index) => {
            props.onChangeStep && props.onChangeStep({ originalEvent: event, index });
            
            // allow user to mark event.preventDefault() to stop this step from advancing
            if (event.defeaultPrevented()) {
                return;
                }
                
               // set the new step
               setActiveStepState(index);
```

Reply:
I made a note to add this later. Thanks for the feedback.